### PR TITLE
fix(debian): take installed files from the origin layer

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Replace https://github.com/aquasecurity/trivy/pull/5844 and https://github.com/aquasecurity/trivy/pull/6682

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/5857

## Related PRs
- #5844 
- #6682

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by DmitriyLewen** [comment]
@knqyf263 I added test - https://github.com/aquasecurity/trivy/pull/6849/commits/bfb713e0dd7e1df190d561f16c7304109429c93c
The rest looks good to me!

**Comment by knqyf263** [comment]
I'll merge this once you approve it.

**Review by DmitriyLewen — APPROVED** [review]

**Pull request merged**
